src/node_contextify.cc

```diff
@@ -269,7 +269,10 @@ class ContextifyContext {
       CHECK(!debug_context.IsEmpty());
     }
     Environment* env = Environment::GetCurrent(args);
-    ScopedEnvironment env_scope(debug_context, env);
+    const int index = Environment::kContextEmbedderDataIndex;
+    if (!debug_context->GetAlignedPointerFromEmbedderData(index)) {
+      ScopedEnvironment env_scope(debug_context, env);
```

Comment:
Are you sure this is right? `env_scope`'s destructor will get called as soon as execution moves outside of the `if` statement. I'm not sure, but we may need to have an `if-else` and copy the code after the current `if` into both blocks (one with `env_scope` and one without)?